File: solutions/2610. Convert an Array Into a 2D Array With Conditions/Solution.ts
function findMatrix(nums: number[]): number[][] {
  const hash = new Set();
  const res: number[][] = [];

  for (let item of nums) {
    if (!(item in hash)) hash[item] = 0;
    const row = hash[item];

    if (res.length === row) res.push([]);
    res[row].push(item);
    hash[item]++;
  }

  return res;
}

File: solutions/2610. Convert an Array Into a 2D Array With Conditions/Solution.py
from collections import Counter
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        hash = Counter(nums)
        res = []
        for item, occurance in hash.items():
            for i in range(occurance):
                if len(res) <= i:
                    res.append([])
                res[i].append(item)
        return res